aiosmtpd/tests/test_smtp.py

```diff
@@ -1659,6 +1659,13 @@ def test_mail_invalid_body_param(self, plain_controller, client):
         resp = client.docmd("MAIL FROM: <anne@example.com> BODY=FOOBAR")
         assert resp == S.S501_MAIL_BODY
 
+    def test_limitlocalpart(self, plain_controller, client):
+        plain_controller.smtpd.local_part_limit = 64
+        client.ehlo("example.com")
+        locpart = "a" * 65
+        resp = client.docmd(f"MAIL FROM: {locpart}@example.com")
+        assert resp == S.S553_MALFORMED
+
```

Comment:
Well, ultimately it will hit the `command_size_limit` of 512 (or less).

So, the longest possible email address (local+domain) without triggering other errors would be about 502 (`512 - "RCPT TO:" - "\r\n"`), or 500 for "MAIL FROM:"

I'll add tests for these 2 extremes.

Reply:
Tests added.